qutil.c

```diff
@@ -310,3 +311,39 @@ args_free(struct args *args) {
 	free(args->buf);
 	free(args);
 }
+
+void
+qlog_func(int pri, int do_errno, const char *func, int lineno,
+    const char *fmt, ...)
+{
+	va_list		 ap;
+	char		*nfmt;
+	int		 saved_errno;
+
+	if (pri > quark_verbose)
+		return;
+
+	if (do_errno)
+		saved_errno = errno;
+
+	va_start(ap, fmt);
+
+	/* best effort in out of mem situations */
+	if (asprintf(&nfmt, "%s:%d: %s", func, lineno, fmt) == -1) {
+		fprintf(stderr, "%s: %s:%d: ",
+		    program_invocation_short_name, func, lineno);
+		vfprintf(stderr, fmt, ap);
+		if (do_errno)
+			fprintf(stderr, "%s:", strerror(saved_errno));
```

Comment:
why the trailing `:`?

Reply:
Cause I'm a tool :rofl:, I'll fix it 
You have good eyes!